### ISSUE TITLE
Use of "external" types

### ISSUE THREAD

**Issue by kraftwer1** [ionic-team/stencil#850]
**Stencil version:**

```
 @stencil/core@0.9.7
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or https://stencil-worldwide.slack.com

**Current behavior:**
Defining and exporting types gives a compilation error:

**my-component.tsx**

    export type Color = "black" | "red";
    // ...
    @Prop() color: Color;

... results in:
    
    [ ERROR ]  typescript: src/components.d.ts, line: 31
    Cannot find name 'Color'.

**Expected behavior:**
No error occurs.

**Steps to reproduce:**
https://github.com/kraftwer1/stencil-component-starter

**Comment by codinronan** [comment]
@kraftwer1 as always, the best way to figure out what is going on here is the [@ionic/core](https://github.com/ionic-team/ionic/blob/master/core/src/components/app/app.tsx) codebase. If you look at `ion-app` you can see they've done the same thing, but differently: Interfaces and types are defined externally, at the top level of the app/library hierarchy, and imported into the component. 

The Stencil compiler (well - Rollup) will see the import and make sure it is included in the generated `components.d.ts` file; whereas an export from the component won't receive the same treatment (it is a candidate for minification and in fact - should be, you don't use it anywhere else in your branch).

I cloned your branch, made that change, and everything worked beautifully. Give it a try. (create `src/interfaces.ts`, define the same type there, import it into `my-component.ts`.)

To the stencil team, this probably belongs in the docs although it doesn't have much to do with Stencil itself, it's more about how Rollup works.

**Comment by adrianschmidt** [comment]
> To the stencil team, this probably belongs in the docs although it doesn't have much to do with Stencil itself, it's more about how Rollup works.

Yes, _please_ document this. I needed to export interfaces to make them externally available, and it took me a lot of searching before I found this.
